Homework5/CompanyVacations/CompanyVacationInfo.cs

```diff
@@ -0,0 +1,91 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace CompanyVacations
+{
+    public class CompanyVacationInfo
+    {
+        private List<EmployeeVacations> allVacationsRecords = new List<EmployeeVacations>();
+
+        public void AddEmployeeVacations(EmployeeVacations x)
+        {
+            if (x is null)
+            {
+                throw new ArgumentNullException();
+            }
+
+            if (AlreadyOnVacation(x))
+            {
+                throw new ArgumentException("Employee is already on vacation");
+            }
+
+            allVacationsRecords.Add(x);
+        }
+
+        public bool AlreadyOnVacation(EmployeeVacations a)
+        {
+            var r = allVacationsRecords.Where(x => x.Name.Equals(a.Name))
+                                       .Where(x => a.VacationsStart <= x.VacationsEnd &&
+                                                    a.VacationsEnd >= x.VacationsStart)
+                                       .Count();
+
+            return r == 0 ? false : true;
+        }
+
+        public double AverageVacationLength()
+        {
+            return allVacationsRecords.Distinct()
+                                      .Select(x => x.VacationsTaken.TotalDays)
+                                      .Average();
+        }
+
+        public IEnumerable<(string, double)> GetAverageOfEachEmployee()
+        {
+            return allVacationsRecords.Distinct().GroupBy(x => x.Name)
+                                      .Select(x => (x.Key, x.Select(c => c.VacationsTaken.TotalDays).Average()))
+                                      .ToList();
+        }
+
+        //method returns first int as a month, second int as a number of employees
+        public IEnumerable<(int, int)> EmployeesPerMonth()
+        {
+            var vacationStartMonths = allVacationsRecords.Select(x => x.VacationsStart.Month);
+
+            var vacationsEndMonths = allVacationsRecords.Where(x => x.VacationsStart.Month != x.VacationsEnd.Month)
+                                         .Select(x => x.VacationsEnd.Month);
+
+            return vacationStartMonths.Concat(vacationsEndMonths).GroupBy(x => x)
+                                    .Select(x => (x.Key, x.Count()))
```

Comment:
Employee may have multiple vacations at the same month. 
Current method returns not a count of employees, but count of vacations, please update